src/sentry/monitors/serializers.py

```diff
@@ -6,22 +6,65 @@
 from sentry.api.serializers import ProjectSerializerResponse, Serializer, register, serialize
 from sentry.models import Project
 
-from .models import Monitor, MonitorCheckIn
+from .models import Monitor, MonitorCheckIn, MonitorEnvironment
+
+
+@register(MonitorEnvironment)
+class MonitorEnvironmentSerializer(Serializer):
+    def serialize(self, obj, attrs, user):
+        return {
+            "status": obj.get_status_display(),
+            "name": obj.environment.name,
+            "lastCheckIn": obj.last_checkin,
+            "nextCheckIn": obj.next_checkin,
+            "dateCreated": obj.monitor.date_added,
+        }
+
+
+class MonitorEnvironmentSerializerResponse(TypedDict):
+    name: str
+    status: str
+    dateCreated: datetime
+    lastCheckIn: datetime
+    nextCheckIn: datetime
 
 
 @register(Monitor)
 class MonitorSerializer(Serializer):
-    def get_attrs(self, item_list, user):
+    def __init__(self, environments=None):
+        self.environments = environments
+
+    def get_attrs(self, item_list, user, **kwargs):
         # TODO(dcramer): assert on relations
         projects = {
-            d["id"]: d
-            for d in serialize(
+            p["id"]: p
+            for p in serialize(
                 list(Project.objects.filter(id__in=[i.project_id for i in item_list])), user
             )
         }
 
+        monitor_environments = {}
+        if self.environments:
+            monitor_environments = {
+                str(item.id): {
+                    me.pop("name"): me
+                    for me in serialize(
+                        list(
+                            MonitorEnvironment.objects.filter(
+                                monitor=item, environment__in=self.environments
+                            )
+                        ),
+                        user,
+                    )
+                }
+                for item in item_list
```

Comment:
This will make a query to `MonitorEnvironment` per `Monitor`, probably better to batch fetch the monitor envs

Reply:
gotcha